test/math.jl

```diff
@@ -249,10 +250,12 @@ end
             @test isinf(log1p(-one(T)))
             @test isnan_type(T, log1p(convert(T,NaN)))
             @test_throws DomainError log1p(convert(T,-2.0))
+            @test hypot(-2) == 2
             @test hypot(T(0), T(0)) === T(0)
             @test hypot(T(Inf), T(Inf)) === T(Inf)
             @test hypot(T(Inf), T(x)) === T(Inf)
-            @test hypot(T(Inf), T(NaN)) === T(Inf)
```

Comment:
@test hypot(T(Inf), T(NaN)) === T(Inf) looked strange. Anyone knows why this test is here? Shouldn't this return NaN?

Reply:
From the C-2011 spec:
> hypot(±∞, y) returns +∞, even if y is a NaN.

Reply:
And the IEEE754-2008 spec:
> For the hypot function, hypot(±0, ±0) is +0, hypot(±∞, qNaN) is +∞, and hypot(qNaN, ±∞) is +∞.

Reply:
I actually don't remember adding these test but as @simonbyrne just pointed out on Slack, `Inf` is following the standard. The reasoning is that `NaN` is used when it cannot be determined if the limit is `0` or `+-Inf`. In either case, the result of `hypot` is `Inf` when the other input is `+-Inf` so returning `Inf` here makes sense.